install/kubernetes/cilium/templates/_helpers.tpl

```diff
@@ -97,3 +97,34 @@ Return the appropriate apiVersion for podDisruptionBudget.
 {{- print "policy/v1beta1" -}}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Generate TLS CA for Cilium
+Note: Always use this template as follows:
+    {{- $_ := include "cilum.ca.setup" . -}}
+
+The assignment to `$_` is required because we store the generated CI in a global `commonCA`
+and `commonCASecretName` variables.
+
+*/}}
+{{- define "cilum.ca.setup" }}
+  {{- if not .commonCA -}}
+    {{- $ca := "" -}}
+    {{- $secretName := "cilium-ca" -}}
+    {{- $crt := .Values.tls.ca.cert -}}
+    {{- $key := .Values.tls.ca.key -}}
+    {{- if and $crt $key }}
+      {{- $ca = buildCustomCert $crt $key -}}
+    {{- else }}
+      {{- with lookup "v1" "Secret" .Release.Namespace $secretName }}
+        {{- $crt := index .data "ca.crt" }}
+        {{- $key := index .data "ca.key" }}
+        {{- $ca = buildCustomCert $crt $key -}}
+      {{- else }}
+        {{- $validity := (max .Values.hubble.tls.auto.certValidityDuration .Values.clustermesh.apiserver.tls.auto.certValidityDuration | int) -}}
```

Comment:
nit: I think we should introduce `tls.ca.certValidityDuration` (or `tls.auto.ca.certValidityDuration`) and use it here without fallback on the existing values.

Reply:
oh good spot :100: